source/common/http/http3/quic_codec_factory.h

```diff
@@ -31,5 +33,43 @@ class QuicHttpClientConnectionFactory : public Config::UntypedFactory {
   std::string category() const override { return "envoy.quic_server_codec"; }
 };
 
+// A factory to create EnvoyQuicClientConnection instance for QUIC
+class QuicClientConnectionFactory : public Config::UntypedFactory {
+public:
+  ~QuicClientConnectionFactory() override = default;
+
+  virtual std::unique_ptr<Network::ClientConnection>
+  createQuicNetworkConnection(Network::Address::InstanceConstSharedPtr server_addr,
+                              Network::Address::InstanceConstSharedPtr local_addr,
+                              Network::TransportSocketFactory& transport_socket_factory,
+                              Stats::Scope& stats_scope, Event::Dispatcher& dispatcher,
+                              TimeSource& time_source) PURE;
+
+  std::string category() const override { return "envoy.quic_connection"; }
+};
```

Comment:
This is a good idea to statically link in EnvoyQuicClientConnection! Maybe the H3 downstream integration test can use this way to make HttpIntegrationTest not to depend on QUICHE.